app/scripts/views/components/dataTable/tableView.js

```diff
@@ -100,43 +100,47 @@
         } else {
           _.each(_selectedSamples, function(caseId) {
             var caseIndex_ = caseIndices[caseId];
-            var caseData_ = data_[caseIndex_];
-            var tempData_ = '';
-            switch (type_) {
-              case 'mutatedGene':
-                tempData_ = caseData_.mutated_genes;
-                includeMutationCount = true;
-                break;
-              case 'cna':
-                tempData_ = caseData_.cna_details;
-                includeMutationCount = false;
-                break;
-              default:
-                var category = caseData_[attributes_.attr_id];
-                if (!category) {
-                  category = 'NA';
-                }
-                if (!selectedMap_.hasOwnProperty(category)) {
-                  selectedMap_[category] = [];
-                }
-                selectedMap_[category].push(caseId);
-                break;
-            }
-            if (isMutatedGeneCna) {
-              _.each(tempData_, function(geneIndex) {
-                if (selectedMap_[geneIndex] === undefined) {
-                  selectedMap_[geneIndex] = {};
-                  if (includeMutationCount) {
-                    selectedMap_[geneIndex].num_muts = 1;
+            if (_.isNumber(caseIndex_)) {
```

Comment:
Originally I thought `caseId` should always have a `caseIndex`, but for some studies (maybe because the data is not curated correctly at server-side), this is incorrect. I saw this issue while testing some other fixes. `caseIndex` may be `undefined`